runtime/bigarray.c

```diff
@@ -382,7 +382,9 @@ CAMLexport int caml_ba_compare(value v1, value v2)
   case CAML_BA_CAML_INT:
   case CAML_BA_NATIVE_INT:
     DO_INTEGER_COMPARISON(intnat);
-  default: CAMLunreachable();
+  default:                      /* Should not happen */
+    CAMLunreachable();
+    return 0;                   /* Keeps GCC -Wall happy */
```

Comment:
Maybe mention `GCC 9` rather than simply `GCC`, so that next time someone wonders whether this line is still needed, that person can check whether gcc 9 is still in use somewhere and able to build the current OCaml codebase.